packages/target-decisioning-engine/src/contextProvider.js

```diff
@@ -91,8 +92,22 @@ export function createMboxContext(mboxRequest) {
 /**
  * @param { import("@adobe/target-tools/delivery-api-client/models/Geo").Geo } geo context
  */
-export function createGeoContext(geoContext = {}) {
-  //TODO
+export async function createGeoContext(geoContext = {}) {
+  // When ipAddress is the only geo value passed in to getOffers(), do IP-to-Geo lookup.
+  if (
+    geoContext.ipAddress &&
+    !geoContext.latitude &&
+    !geoContext.longitude &&
+    !geoContext.countryCode &&
+    !geoContext.stateCode &&
+    !geoContext.city
+  ) {
+    const ipToGeoLookupResponse = await fetch(
+      `https://assets.adobetarget.com/v1/geo/`
```

Comment:
you will want to use `getCdnBasePath` here rather than hard code te full url. see https://github.com/jasonwaters/target-nodejs-sdk/blob/master/packages/target-decisioning-engine/src/utils.js#L124

Reply:
Updated